### PR TITLE
fix #8818 - run regendoc without tox cachedir

### LINKED ISSUE #8818: Investigate long regendoc output in cache.rst
Hmmm with the documentation moved we are getting cache data from other runs. @RonnyPfannschmidt can we use "regendoc wipe" to fix this?

_Originally posted by @nicoddemus in https://github.com/pytest-dev/pytest/pull/8814#discussion_r660068177_

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
#4270 introduced per tox env pytest cachedirs and
never protected regendoc against it

closes #8818

### CODE REVIEW

File: changelog/8818.trivial.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Ensure regendoc opts out of TOX_ENV cachedir selection to ensure indeependent example test runs.
```
Comment (jugmac00) [inline changelog/8818.trivial.rst:1]:
```suggestion
Ensure regendoc opts out of TOX_ENV cachedir selection to ensure independent example test runs.
```

File: doc/en/how-to/usage.rst [hunk 2]
```diff
@@ -201,28 +201,8 @@ hook was invoked:
 .. code-block:: pytest
 
     $ python myinvoke.py
-    .FEsxX.                                                              [100%]*** test run reporting finishing
+    *** test run reporting finishing
 
-    ================================== ERRORS ==================================
```
Comment (nicoddemus) [inline doc/en/how-to/usage.rst:206]: Strange, why was this block removed? 🤔

**Review by nicoddemus — COMMENTED** [review]
Thanks for working on this @RonnyPfannschmidt. Left a few comments.

File: doc/en/Makefile [hunk 3]
```diff
@@ -34,6 +34,10 @@ REGENDOC_ARGS := \
 
 regen: REGENDOC_FILES:=*.rst */*.rst
 regen:
-	PYTHONDONTWRITEBYTECODE=1 PYTEST_ADDOPTS="-pno:hypothesis -Wignore::pytest.PytestUnknownMarkWarning" COLUMNS=76 regendoc --update ${REGENDOC_FILES} ${REGENDOC_ARGS}
+# need to reset cachedir to the non-tox default
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply doc/en/Makefile:37]: The `-o cache_dir` option was added

File: doc/en/builtin.rst [hunk 4]
```diff
@@ -28,48 +33,49 @@ For information about fixtures, see :ref:`fixtures`. To see a complete list of a
 
         Values can be any object handled by the json stdlib module.
 
-    capsys
+    capsys -- ../../../..$PYTHON_SITE/_pytest/capture.py:903
```
[the remark this answers is not in the document]
Comment (RonnyPfannschmidt) [reply doc/en/builtin.rst:36]: It's a bug in the new feature, I'd consider the fix a follow-up atm

File: doc/en/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicoddemus) [reply doc/en/Makefile:37]: Ahh right sorry for the noise! 🤦

File: doc/en/builtin.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (nicoddemus) [reply doc/en/builtin.rst:36]: Fair enough! Could you create an issue so we don't lose sight of it then? Thanks!

File: doc/en/Makefile [hunk 5]
```diff
@@ -34,6 +34,10 @@ REGENDOC_ARGS := \
...
+	PYTHONDONTWRITEBYTECODE=1 \
+	PYTEST_ADDOPTS="-pno:hypothesis -p no:hypothesispytest -Wignore::pytest.PytestUnknownMarkWarning -o cache_dir=.pytest_cache" \
```
Comment (The-Compiler) [inline doc/en/Makefile:39]: Where's the `-p no:hypothesispytest` coming from?

File: doc/en/builtin.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (The-Compiler) [reply doc/en/builtin.rst:36]: We already have one: #8822

File: tox.ini [hunk 6]
```diff
@@ -90,17 +90,14 @@ install_command=python -m pip --use-deprecated=legacy-resolver install {opts} {p
...
     sphinx
 whitelist_externals =
     rm
     make
 commands =
     # don't show hypothesis plugin info in docs, see #4602
-    pip uninstall hypothesis -y
-    rm -rf /tmp/doc-exec*
-    rm -rf {envdir}/.pytest_cache
-    make regen
```
Comment (The-Compiler) [inline tox.ini:103]: @RonnyPfannschmidt If you removed those, I suppose the comment should go away too, no?

File: doc/en/how-to/usage.rst (hunk 2 above)
Comment (The-Compiler) [reply doc/en/how-to/usage.rst:206]: Looking at it in the context of the text above, I think the old output was wrong (other files leaking in somehow?) and the new one seems correct, no?

File: doc/en/Makefile (hunk 5 above)
Comment (RonnyPfannschmidt) [reply doc/en/Makefile:39]: New plugin entry point of hypothesis

File: tox.ini (hunk 6 above)
Comment (RonnyPfannschmidt) [reply tox.ini:103]: done, thanks

**Review by nicoddemus — APPROVED** [review]

**Pull request merged**